src/components/CreateEventForm/CreateEventForm.tsx

```diff
@@ -104,12 +107,15 @@ export default function CreateEventForm(
         // eslint-disable-next-line
         console.log('USER_CREATE_EVENT', formValue);
         if (setFormHook) setFormHook(formValue);
-        createEvent(formValue)
-            .then((eventId) => {
-                history.push(`/event/${eventId}`);
-            })
+
+        try {
+            await recaptchaRef.current!.executeAsync();
```

Comment:
What's the benefit to using the async option here over sync?

Reply:
Verification? Like verifying if the user is a bot or not? We're using invisible recaptcha so it doesn't need the user to answer any challenges unless it believes that you're a bot. 
We kept getting errors when using sync over async

Reply:
Went async here because just didnt want to have to deal with callback hell. 
```
recaptchaRef.current!.execute()
  .then(token => {
    submitForm()
      .then()
  })
  .catch()
```

Also just to make it obvious that we need to make sure this recaptcha executes before the form is actually submitted.

Reply:
`execute()` has a return type of void so there is no promise to handle.

I was initially thinking the need to have the async was to send the token in with the form but if the plan is to handle that separately we could just use the sync version since await makes the `executeAsync()` sync anyways. What do you think?

Reply:
When I was using `execute` there would be times when the response token was not being appended to the form and I think this is because there is some time needed to generate the response token. So went with `executeAsync` to guarantee that the token is generated and appends to the form. 

Maybe we should append the token to the form ourselves and then also do a guard

```
const token = await recaptchaRef.current!.executeAsync();
if (!token) return; // cancel form submission and or display error
```